plugins/http/src/commands.rs

```diff
@@ -264,6 +264,14 @@ pub async fn fetch<R: Runtime>(
                     }
                 }
 
+                // In case empty origin is passed, remove it. Some services do not like Origin header
+                // so this way we can remove it in explicit way. The default behaviour is still to set it
+                if cfg!(feature = "unsafe-headers")
+                    && headers.get(header::ORIGIN) == Some(&HeaderValue::from_static(""))
```

Comment:
I don't think we should match on an empty string, because there might be some users who want to send empty string for `Origin`. I'd rather add a new option when making the request, for example:
```js
fetch(url, {
  sendOriginHeader: false
});
```

Reply:
This is something I was thinking a lot. Like I'd like to avoid adding extra parameters to the fetch and try to keep the fetch as close to global fetch. And in case of patching global fetch with tauri-http (example to get control what fetches actually can be made) the consumers assume standard fetch. Different libraries support some pre/post hooks for different operations that allow example inject headers and such like for custom auth and they operate with the `RequestInit` object.

And the syntax of Origin header is:
```
Origin: null
Origin: <scheme>://<hostname>
Origin: <scheme>://<hostname>:<port>
```
[MDN origin header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Origin)
[RFC 6454](https://datatracker.ietf.org/doc/html/rfc6454#section-6.1)

Sending an empty string is invalid format. In case user wants to hide the origin for example privacy reasons they should set it to `null` string literal. 

For these reasons I ended up with sending empty string in the origin header as the mechanism.

I hope my reasoning makes sense.

Reply:
I was comparing with how node and deno runtimes behave and both allow setting `Origin` to an empty string. However your reasoning is quite valid and I think we should go with your approach until someone else asks for this specific behavior.